Remove code from Compiler that is not used anymore

alalazo opened the pull request:
#45784 unified the search mechanism of `spack compiler find` and `spack external find`. This PR removes attributes from `Compiler` classes that were mainly used in the old search mechanism.

**Prerequisites**

- [x] #46037
- [x] #46038

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

alalazo committed:
Remove Compiler.cc_names

alalazo committed:
Remove Compiler.f77_names

alalazo committed:
Remove Compiler.f77_version

alalazo committed:
Remove Compiler.fc_version, except for msvc

hainest commented on var/spack/repos/builtin/packages/boost/package.py:
```diff
@@ -462,24 +462,24 @@ def flag_handler(self, name, flags):
-            "FCC": "clang",
+            "%gcc": "gcc",
+            "%intel": "intel",
+            "%oneapi": "intel",
```
I'm not sure if Boost builds with the oneapi compiler. I'll have to double-check.

hainest commented on var/spack/repos/builtin/packages/boost/package.py:
```diff
@@ -462,24 +462,24 @@ def flag_handler(self, name, flags):
 
         for cc, toolset in toolsets.items():
-            if cc in self.compiler.cxx_names:
+            if cc in self.spec.satiisfies(cc):
```
1. Typo in 'satisfies'
2. Should this just be `if self.spec.satisfies(cc)`?

alalazo replied:
Thanks for catching this typo. I was planning to extract package changes from this PR and ping people there.

alalazo committed:
Remove Compiler.cxx_version

alalazo committed:
Remove Compiler.cxx_names

alalazo committed:
Remove Compiler.fc_names

alalazo committed:
molgw: remove Compiler.fc_names

alalazo committed:
Mark a test that is failing on Windows

The functionality is never used, test is failing because
spack.compilers.is_mixed_toolchain is now based on detection.

Signed-off-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

alalazo committed:
Mark test as not for Win

Signed-off-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

alalazo commented on lib/spack/spack/test/architecture.py:
```diff
@@ -130,6 +130,7 @@ def test_arch_spec_container_semantic(item, architecture_str):
     ],
 )
 @pytest.mark.filterwarnings("ignore:microarchitecture specific")
+@pytest.mark.not_on_windows("Windows doesn't support the compiler wrapper")
```
One of two tests I had to disable on Windows

alalazo commented on lib/spack/spack/test/compilers/basics.py:
```diff
@@ -589,6 +589,7 @@ def test_xl_r_flags():
     "compiler_spec,expected_result",
     [("gcc@4.7.2", False), ("clang@3.3", False), ("clang@8.0.0", True)],
 )
+@pytest.mark.not_on_windows("GCC and LLVM currently not supported on the platform")
```
The other one

alalazo reviewed (commented):
I had to disable 2 tests on Windows, because the logic for determining if a toolchain is "mixed" changes here. 

I don't think Windows is affected in any way[^1], but I'll let @johnwparent comment on that.

[^1]: The `is_mixed_toolchain` function is used to inject compiler flags via the wrapper, which Windows is not using afaik

johnwparent commented:
Already merged, but FWIW @alalazo this is perfectly fine from the Windows side. All the flag injection logic will require a refactor anyway once the Windows compiler wrapper lands, so perfectly fine to skip/remove whatever you need from Windows for the time being. Appreciate the ping.

alalazo commented:
Now that pipelines are back, this seems to be causing failures on aws-pcluster. I'll investigate asap.

alalazo commented:
https://github.com/spack/spack/pull/46505
